docs/guides/cache/registry.md

```diff
@@ -0,0 +1,106 @@
+  the final stage.
+- It works with other exporters for more flexibility, instead of only the
+  `image` exporter.
+
+> **Note**
+>
+> The `registry` cache storage backend requires using a different driver than
+> the default `docker` driver - see more information on selecting a driver
+> [here](../drivers/index.md). To create a new docker-container driver (which
+> can act as a simple drop-in replacement):
+>
+> ```console
+> docker buildx create --use --driver=docker-container
+> ```
+
+To import and export your cache using the `registry` storage backend we use the
+`--cache-to` and `--cache-from` flags and point it to our desired image target
+using the `ref` parameter:
+
+```console
+$ docker buildx build --push -t <user>/<image> \
+  --cache-to type=registry,ref=<user>/<cache-image> \
+  --cache-from type=registry,ref=<user>/<cache-image> .
+```
+
+You can choose any valid reference value for `ref`, as long as it's not the
+same as the target location that you push your image to. You might choose
+different tags (e.g. `foo/bar:latest` and `foo/bar:build-cache`), separate
+image names (e.g. `foo/bar` and `foo/bar-cache`), or even different
+repositories (e.g. `docker.io/foo/bar` and `ghcr.io/foo/bar`).
+
+If the cache does not exist, then the cache import step will fail, but the
+build will continue.
+
+## Cache options
+
+Unlike the simple `inline` cache, the `registry` cache has lots of parameters to
+adjust its behavior.
+
+### Cache mode
+
+Build cache can be exported in one of two modes: `min` or `max`, with either
+`mode=min` or `mode=max` respectively. For example, to build the cache with
+`mode=max`:
+
+```console
+$ docker buildx build --push -t <user>/<image> \
+  --cache-to type=registry,ref=<user>/<cache-image>,mode=max \
+  --cache-from type=registry,ref=<user>/<cache-image> .
+```
+
+Note that only `--cache-to` needs to be modified, as `--cache-from` will
+automatically extract the relevant parameters from the resulting output.
+
+In `min` cache mode (the default), only layers that are exported into the
+resulting image are cached, while in `max` cache mode, all layers are cached,
+even those of intermediate steps.
+
+While `min` cache is typically smaller (which speeds up import/export times,
+and reduces storage costs), `max` cache is more likely to get more cache hits.
+Depending on the complexity and location of your build, you should experiment
+with both parameters to get the results.
+
+### Cache compression
+
+Since `registry` cache is exported separately from the main build result, you
+can specify separate compression parameters for it (which are similar to the
+options provided by the `image` exporter). While the defaults have been
+selected to provide a good out-of-the-box experience, you may wish to tweak the
+parameters to optimize for storage vs compute costs.
+
+To select the compression algorithm, you can use the `compression=<uncompressed|gzip|estargz|zstd>`
```

Comment:
does this require a `oci-mediatypes=true`? https://github.com/moby/buildkit#:~:text=estargz%20should%20be%20used%20with%20oci%2Dmediatypes%3Dtrue

Reply:
`oci-mediatypes` will be implicitly set in these cases for estargz. I'm not sure if we need to call this out here though, I'm on the fence about how much to discuss `oci-mediatypes`, since ideally, at some point we'd switch to using this as the default.

Reply:
I've gone and added a note about the `oci-mediatypes` option for the `registry` and `local` exporters, but left off the details about `estargz` for now. Later, we'll cross-link to `../exporters/image.md` (which will have much more information on that interaction) once I get round to writing those :smile: